haystack/nodes/retriever/base.py

```diff
@@ -126,12 +126,15 @@ def eval(
         # Collect questions and corresponding answers/document_ids in a dict
         question_label_dict = {}
         for label in labels:
-            id_question_tuple = (label.document_ids[0], label.query)
-            if open_domain:
-                question_label_dict[id_question_tuple] = [a for a in label.answers if a is not None]
-            else:
-                deduplicated_doc_ids = list(set([str(x) for x in label.document_ids]))
-                question_label_dict[id_question_tuple] = deduplicated_doc_ids
+            # document_ids are empty if no_answer == True
+            if not label.no_answer:
+                id_question_tuple = (label.document_ids[0], label.query)
+                if open_domain:
```

Comment:
Let's double-check how we want to call this. Maybe we can come up with a self-explaining term.